Nos/Views/Moderation/UserFlagView.swift

```diff
@@ -0,0 +1,137 @@
+import SwiftUI
+
+/// Displays pickers for selecting user flag options, with additional stages shown
+/// based on previous selections.
+struct UserFlagView: View {
+    @Binding var selectedFlagOption: FlagOption?
+    @Binding var selectedSendOption: FlagOption?
+    @Binding var selectedVisibilityOption: FlagOption?
+    @Binding var showSuccessView: Bool
+
+    /// The target of the report.
+    let flagTarget: ReportTarget
+    var sendAction: () -> Void
+
+    @Environment(\.dismiss) private var dismiss
+
+    @State private var flagCategories: [FlagOption] = []
+
+    var body: some View {
+        ZStack {
+            Color.appBg.ignoresSafeArea()
+            Group {
+                if showSuccessView {
+                    FlagSuccessView()
+                } else {
+                    categoryView
+                }
+            }
+            .padding()
+            .nosNavigationBar(title: .localizable.flagUserTitle)
+            .toolbar {
+                ToolbarItem(placement: .navigationBarLeading) {
+                    Button(action: {
+                        dismiss()
+                        resetSelections()
+                    }, label: {
+                        Text(.localizable.cancel)
+                            .foregroundColor(.primaryTxt)
+                    })
```

Comment:
It would be nice to use the slightly shorter syntax for creating buttons:

```suggestion
                    Button {
                        dismiss()
                        resetSelections()
                    } label: {
                        Text(.localizable.cancel)
                            .foregroundColor(.primaryTxt)
                    }
```